DigitalLearningSolutions.Web/Controllers/LearningPortalController/Current.cs

```diff
@@ -165,5 +167,38 @@ public async Task<IActionResult> LaunchLearningResource(int learningLogItemId)
             // TODO: HEEDLS-678 redirect user to new LH forwarding endpoint.
             return Redirect(learningResourceLink);
         }
+
+        [HttpGet]
+        [ServiceFilter(typeof(VerifyLearningLogItemExists))]
+        [Route("/LearningPortal/Current/MarkAsComplete/{learningLogItemId:int}")]
+        public IActionResult MarkCurrentLearningLogItemAsComplete(int learningLogItemId)
+        {
+            var learningLogItem =
+                learningLogItemsService.SelectLearningLogItemById(learningLogItemId);
+
+            var model = new SetLearningLogItemCompletionDateViewModel(learningLogItemId, learningLogItem!.Activity!);

```

Comment:
Are we OK with the assumption that Activity is not null here?

Reply:
I think so? The column is nullable but I would assume a user would never encounter a resource without a name

Reply:
We're saving to that column when we add a resource to their action plan, so I would assume not. I guess it will be OK.

Reply:
I think we actually need to do things in a slightly different way. Because the details for the resources are stored on a different database, we can't guarantee that whatever we have saved here is up to date.

We've gone with a more safe-than-sorry approach in other tickets (671 is probably the closest example, if it gets merged first a method that it creates can probably be reused here) where we grab the resource reference Id and call the API every time we want to display any information from it to ensure it is up to date.

Reply:
Adjusted this to the same way it's done in 671